package_managers/apt_key.bzl

```diff
@@ -0,0 +1,51 @@
+#Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for configuring apt GPG keys"""
+
+load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build")
+load("//util:run.bzl", "container_run_and_extract")
+
+def add_apt_key(name, key, image):
+    initial_image = "//package_managers:apt_key_image"
```

Comment:
//package_managers:apt_key_image won't be available from another repo. And the key extracted based on this image might not be compatible with arbitrary images. What about leaving it as an input to the rule instead of  not hardcoding this?

I'm thinking if we could change the add_apt_key rule to do all of the following: 
- install gnupg in the base image,
- run the apt-key add command, 
- extract the key file,
- and output the image which is base+key file.

I'm not 100% sure if this will cause determinism issue. Since these are all happen before download_pkgs rule, and we are going to cache the output of download_pkgs to make install_pkgs deterministic. I guess it might be ok? WDYT?

Reply:
> install gnupg in the base image,
> run the apt-key add command,

To make sure I understand, this would install gnupg in the base image, and then abandon that image? 

That should work, but like you pointed out it won't technically be deterministic, since we can't use the output of download_pkgs here, since it's technically "relative" to the base image.

I'll have to think on it a bit to see if anything could go wrong.

Reply:
Yea. Install gnupg in the base, run the apt-get command, extract the key, abandon this image, and output the key+a clean base image.